### PR TITLE
docs: Fix nested list

### PR DESCRIPTION
**Pull request by itsdonnix**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Fix nested lists

### CODE REVIEW

**Comment by Xerkus** [comment]
Lists are already valid and correctly formatted with 2 spaces indentation.

Markdown spec allows up to 3 extra leading spaces that are ignored and stripped. We, however, do not allow them for uniformity.

**Comment by froschdesign** [comment]
@Xerkus 
But for some reason, MkDocs only works with 4 spaces.

Some background: https://github.com/Python-Markdown/markdown/issues/3

**Comment by Xerkus** [comment]
@froschdesign I found https://github.com/mkdocs/mkdocs/issues/2719#issuecomment-1000637764

> looks like I added mdx_truly_sane_lists into the markdown extension and that changed the indent to 2 spaces instead of 4.

Can it be fixed in our docs builder with that?

**Comment by froschdesign** [comment]
@Xerkus 
I'm already searching for it…

**Comment by Xerkus** [comment]
@froschdesign reading your link, I understand mkdocs is not using commonmark?

**Comment by froschdesign** [comment]
MkDocs uses [Python-Markdown](https://python-markdown.github.io):

> ### Indentation/Tab Length
> 
> The [syntax rules](https://daringfireball.net/projects/markdown/syntax#list) clearly state that when a list item consists of multiple paragraphs, “each subsequent paragraph in a list item must be indented by either 4 spaces or one tab” (emphasis added). However, many implementations do not enforce this rule and allow less than 4 spaces of indentation. The implementers of Python-Markdown consider it a bug to not enforce this rule.
> 
> This applies to any block level elements nested in a list, including paragraphs, sub-lists, blockquotes, code blocks, etc. They must always be indented by at least four spaces (or one tab) for each level of nesting.

https://python-markdown.github.io/#differences

File: docs/book/v1/index.md [hunk 1]
```diff
@@ -5,16 +5,16 @@ to provide a minimalist [PSR-7](http://www.php-fig.org/psr/psr-7/) middleware
 framework for PHP, with the following features:
 
 - Routing. Choose your own router; we support:
-  - [Aura.Router](https://github.com/auraphp/Aura.Router)
-  - [FastRoute](https://github.com/nikic/FastRoute)
-  - [laminas-router](https://github.com/mezzio/mezzio-router)
+    - [Aura.Router](https://github.com/auraphp/Aura.Router)
```
Comment (Xerkus) [inline docs/book/v1/index.md]: Will blank line before nested list make linter pass?

**Comment by Xerkus** [comment]
Can't be merged until linter is updated to allow/require 4 space indentation for nested lists as otherwise it will cause CI failures everywhere.

**Comment by Xerkus** [comment]
@itsdonnix markdown lint was updated. Can you please also fix remaining nested lists?

**Comment by itsdonnix** [comment]
> @itsdonnix markdown lint was updated. Can you please also fix remaining nested lists?

Sure @Xerkus ! I forgot to indenting the v1 application section and the readme.

**Comment by itsdonnix** [comment]
I think the CI only checks for the readme file

File: docs/book/v1/features/application.md [hunk 2]
```diff
@@ -217,7 +217,7 @@ methods for retrieving them. They include:
   typically a `Mezzio\Emitter\EmitterStack` instance.
 - `getFinalHandler(ResponseInterface $response = null)`: retrieves the final
   handler instance. This is middleware with the signature `function ($request,
-  $response, $error = null)`, and it is invoked when the middleware pipeline
+$response, $error = null)`, and it is invoked when the middleware pipeline
```
Comment (Xerkus) [inline docs/book/v1/features/application.md]: This is ambiguous change. I am not sure it is entirely correct.
Can you please reformat it to put code span on a single line?

**Comment by Xerkus** [comment]
@itsdonnix  Thank you. I was totally unaware.

**Review by gsteel — APPROVED** [review]
Thanks @itsdonnix

**Pull request merged**